osbuild/util/sbom/spdx2/model.py

```diff
@@ -216,6 +217,60 @@ def to_dict(self):
         }
 
 
+class ExtractedLicensingInfo():
+    """
+    Represents extracted licensing information for a license not on the SPDX License List.
+
+    https://spdx.github.io/spdx-spec/v2.3/other-licensing-information-detected/
+    """
+
+    def __init__(self, extracted_text: str, name: Optional[str] = None) -> None:
+        self.extracted_text = extracted_text
+        self.name = name
+        self.license_ref_id = self._generate_license_id(self.extracted_text, self.name)
+
+    @classmethod
+    def _generate_license_id(cls, extracted_text: str, name: Optional[str] = None) -> str:
+        """
+        Generate a unique SPDX license ID by hashing the extracted text using SHA-256.
+
+        If a license name is provided, include it in the license ID.
+        """
+        extracted_text_hash = hashlib.sha256(extracted_text.encode()).hexdigest()
+        if name is not None:
+            return f"LicenseRef-{cls._normalize_name_for_id(name)}-{extracted_text_hash}"
+        return f"LicenseRef-{extracted_text_hash}"
+
+    @staticmethod
```

Comment:
Let's move this function out of the class, `@staticmethod` is really only a namespace and we're already in one (the module).

Reply:
Its functionality is not affected by the class nor an instance of that class hence to me it doesn't belong in the class. Renaming it `as_license_id` (with its module name being `spdx`, so `spdx.as_license_id(text)`) is clear enough for me.

Not super important either, of course. I just very much dislike `@staticmethod` as to me it's misuse of classes as namespaces when we already have namespaces in modules.

Reply:
done